File: app/src/main/java/com/anandm/scanner/di/ActivityBuilder.java
package com.anandm.scanner.di;

import com.anandm.scanner.BarcodeCaptureActivity;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
abstract public class ActivityBuilder {

    @ContributesAndroidInjector(modules = {ScannerModule.class})
    abstract BarcodeCaptureActivity bindBarcodeCaptureActivity();

}

File: app/src/main/java/com/anandm/scanner/di/ActivityBuilder.kt
package com.anandm.scanner.di

import com.anandm.scanner.BarcodeCaptureActivity
import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class ActivityBuilder {

    @ContributesAndroidInjector(modules = [ScannerModule::class])
    abstract fun bindBarcodeCaptureActivity(): BarcodeCaptureActivity?

}